### ISSUE TITLE
Browsers don't send If-None-Match when requesting entity CSV

### ISSUE THREAD

**Issue by lognaturel** [getodk/central-backend#827]
Reproduce:
- Go to https://test.getodk.cloud/#/projects/149/datasets
- Turn on browser inspection
- Request an entity CSV, see the 200 and that the response includes an Etag header
- Request an entity CSV, see the 200 and that the response includes the same Etag header as before, notice that the request did not include an `If-None-Match` header.

I would have expected a 304 for the second one. I suspect that this has to do with making a direct backend request rather than one through the frontend so this might actually be a frontend issue or maybe one we don't want to address.

**Comment by matthew-white** [comment]
I took a quick look into this, and apparently in Chrome devtools, in the Network tab, there's a checkbox to "Disable cache", which defaults to checked. When I uncheck it, I see `If-None-Match` being sent.

**Comment by lognaturel** [comment]
Very interesting! I'm using Firefox. All other resources I expect to be cached are (e.g. version.txt, javascript, etc).

**Comment by matthew-white** [comment]
I see the same thing in Firefox (also Safari).

> All other resources I expect to be cached are (e.g. version.txt, javascript, etc).

It's pretty rare for Backend itself to return a 304, but I guess nginx adds ETags to many resources and handles the subsequent `If-None-Match`.

I tried downloading an entities.csv locally and saw the same behavior in Chrome and Firefox as I did for the test server: Chrome sent `If-None-Match`, but Firefox did not.

I thought the `Vary` header might somehow be involved in this. But I turned off nginx gzipping of CSVs locally, and while that removed the `Vary` header, it didn't result in `If-None-Match` being sent in Firefox.

When I check about:cache in Firefox, I see entities.csv, but its "data size" is 0 bytes. Not sure whether that's meaningful. Most entries in the cache seem to have a value for "last modified" and "expires", but entities.csv doesn't. I'm not sure that those are needed given that we're using ETags, but I just thought I'd mention it.

I think I'll leave it at here for now and turn back to other tasks. Pretty interesting though!
